c_src/optics_htable/htable.h

```diff
@@ -0,0 +1,88 @@
+/* htable.h
+   Rémi Attab (remi.attab@gmail.com), 10 Mar 2016
+   FreeBSD-style copyright and disclaimer apply
+*/
+
+#pragma once
+
+#include <string.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+// -----------------------------------------------------------------------------
+// struct
+// -----------------------------------------------------------------------------
+
+enum { htable_key_max_len = 1024 };
```

Comment:
This restriction is not really relevant for your use case and should really be `NAME_MAX` or whatever that define is.